### PR TITLE
Switch to standard record format in all editions and tools

### PR DESCRIPTION
**Pull request by MishaDemianenko**
Rename community to standard format.
Update all tools and editions to use standard record format as default.
Introduce `dbms.record_format` setting to officially support record format overriding.

### CODE REVIEW

**Comment by MishaDemianenko** [comment]
@benbc we introduce new public setting here, can you please also take a look.

File: community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java [hunk 1]
```diff
@@ -102,6 +102,10 @@
             "allows an upgrade to be performed.")
     public static final Setting<Boolean> allow_store_upgrade = setting("dbms.allow_format_migration", BOOLEAN, FALSE );
 
+
+    @Description( "Specifies custom storage format name. If not set default format will be used." )
```
Comment (benbc) [inline community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java]: "custom" seems weird here. Also the "name" is probably unnecessary, isn't it? How about "The storage format."?

File: community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java [hunk 2]
```diff
@@ -102,6 +102,10 @@
...
+    public static final Setting<String> record_format = setting( "dbms.record_format", Settings.STRING, "" );
```
Comment (benbc) [inline community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java:107]: I would expect this to be an enum setting (like `ha.branched_data_policy`). Currently the user has no way to know what valid values are.

Comment (benbc) [inline community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java:107]: Why is the default format the default value of the setting?

Comment (MishaDemianenko) [inline community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java:107]: This property should be used only if you want to override default format selection. Default format still edition specific.
Do you think it's make sense to describe it in description or change a name to something like: `dbms.record_format.override`?

**Pull request merged**